libpromises/evalfunction.c

```diff
@@ -7062,6 +7062,27 @@ static FnCallResult FnCallStringSplit(ARG_UNUSED EvalContext *ctx,
 
 /*********************************************************************/
 
+static FnCallResult FnCallStringReplace(ARG_UNUSED EvalContext *ctx,
+                                        ARG_UNUSED const Policy *policy,
+                                        ARG_UNUSED const FnCall *fp,
+                                        const Rlist *finalargs)
+{
+    char *string = RlistScalarValue(finalargs);
+    char *match = RlistScalarValue(finalargs->next);
+    char *substitute = RlistScalarValue(finalargs->next->next);
```

Comment:
Shouldn't you check for NULL before dereferencing pointers? At least an assert or two. Have you tried using `string_replace` with too few args?

Reply:
There is already code which should check that arguments are correct, and ensure that `finalargs->next->next` is valid for this function. But being extra cautios and adding ifs is also fine.